client/dashboard/components/logs-activity/test/activity-event.test.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * @jest-environment jsdom
+ */
+import { render, screen } from '@testing-library/react';
+import { ActivityEvent } from '../activity-event';
+import type { ReactNode } from 'react';
+
+jest.mock( '@automattic/calypso-config', () => jest.fn( () => '' ) );
+jest.mock( '@wordpress/components', () => ( {
+	__experimentalHStack: ( {
+		children,
+		className,
+	}: {
+		children: ReactNode;
+		className?: string;
+	} ) => <div className={ className }>{ children }</div>,
+} ) );
```

Comment:
You can remove this mocking by just adding the following instead:

```suggestion
// Mock scrollIntoView for JSDOM compatibility
Element.prototype.scrollIntoView = jest.fn();
```

This comes frequently enough that I wonder if we should just create a custom test environment that mocks this permanently, for good. That might be a big lift though.

FWIW the test file at `client/dashboard/me/preferences-login/test/index.test.tsx` is the template that I reuse for testing things with minimal mocking. It's difficult to get right so the moment I got it right, I bookmark/templatize it 😆

Reply:
I eventually removed since it was not needed and added some extra test to illustrate the logic ;)